### PR TITLE
KAA-831 [C SDK] Ping timeout should be twice smaller than channel timeout

### PR DESCRIPTION
**Pull request by forGGe**
Fix provided.

### CODE REVIEW

File: client/client-multi/client-c/test/kaa_test.h [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #ifndef KAA_TEST_H_
 #define KAA_TEST_H_
 
-#if defined(KAA_TEST_CUNIT_ENABLED)
+#if defined(KAA_TEST_CUNIT_ENABLED) && 0
```
Comment (ghost) [inline client/client-multi/client-c/test/kaa_test.h]: Is this really need?

File: client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c [hunk 2]
```diff
@@ -1229,6 +1202,40 @@ bool is_service_pending(kaa_tcp_channel_t *self, const kaa_service_t service)
...
+ * array. Destination count is a current amount of items in destination array.
+ *
+ * Returns new array count.
+ */
+static size_t append_with_new_services(kaa_service_t *dest,
+                                       size_t dest_count,
+                                       size_t dest_capacity,
+                                       const kaa_service_t *src,
+                                       size_t src_count)
+{
```
Comment (ohamov) [inline client/client-multi/client-c/src/kaa/platform-impl/common/kaa_tcp_channel.c:1216]: Please add check via KAA_RETURN_IF_NIL(...) for **dest** and **src** parameters

**Comment by ohamov** [comment]
LGTM

**Comment by ghost** [comment]
LGTM

**Pull request merged**